frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -163,7 +315,7 @@ class DashboardInner extends Component {
     this.props.toggleSidebar(SIDEBAR_NAME.addQuestion);
   };
 
-  onMainScroll = event => {
+  onMainScroll = (event: any) => {
```

Comment:
```suggestion
  onMainScroll = (event: Event) => {
```

Reply:
Not sure how we should type this 🤔 `Event` doesn't seem to be aware of the `target` element or about the fact that we're scrolling, so it doesn't have the `event.target.scrollTop` property that we need. Couldn't find a proper type for this and tried to workaround it with an explicit type check, but still no luck:

```ts
if (
  event.target &&
  'scrollTop' in event.target &&
  typeof event.target.scrollTop === "number"
) {
  event.target.scrollTop // still doesn't work 😢
}
```

WDYT?

Reply:
`getMainElement()` returns `HTMLElement` so we can do this:

```tsx
  onMainScroll = (event: Event) => {
    if (event.target instanceof HTMLElement) {
      this.setState({ hasScroll: event.target.scrollTop > 0 });
    }
  };
```

or even:
```tsx
  onMainScroll = (event: Event) => {
    if (!(event.target instanceof HTMLElement)) {
      // sanity check, this should never happen
      throw new Error("main element is not an HTMLElement");
    }

    this.setState({ hasScroll: event.target.scrollTop > 0 });
  };
```